flink-connectors/flink-connector-kafka-0.11/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer011.java

```diff
@@ -174,10 +174,16 @@
 
 	/**
 	 * Descriptor of the transactional IDs list.
+	 * Note: This state is serialized by Kryo Serializer and it has compatibility problem that will be removed later.
```

Comment:
Do we have a plan to deprecate 0.11? This class share a lot of code with FlinkKafkaProducer. After Kafka 0.10.2, the clients can actually talk to any broker at 0.10+. So it may not necessary to keep 0.11 clients anymore. This will help remove quite some duplicate code.

Reply:
We do not have any specific plans. We were waiting to make sure that universal connector is stable and we also need to solve the migration story from `0.11` to the `universal` connector (this PR). So if everything goes well and there won't be any issues with the migration, the soonest we could remove `0.11` connector is `1.10` release.

However I'm hardly an expert in this regard (maintaining public APIs) and I think it's better to coordinate those plans with @aljoscha or @tzulitai .